src/provider/rainbowKit/RainbowKitAccount.ts

```diff
@@ -331,24 +341,26 @@ class RainbowKitAccount extends Eip6963Provider {
     await this.changeNetworkIfRequired("Carbon", network);
   }
 
-  async signEip712(evmHexAddress: string, accountNumber: string, evmChainId: string, msgs: readonly AminoMsg[], fee: StdFee, memo: string, sequence: string, feePayer: string = ''): Promise<string> {
-    await this.verifyNetworkAndConnectedAccount(evmHexAddress, parseChainId(evmChainId))
-    const stdSignDoc = makeSignDoc(msgs, fee, evmChainId, memo, accountNumber, sequence)
+  async signEip712(evmHexAddress: string, accountNumber: string, evmChainId: string, msgs: readonly AminoMsg[], fee: StdFee, memo: string, sequence: string, feePayer: string = ''): Promise<{sig: string, signedDoc: CarbonTx.StdSignDoc}> {
+    const { chainId } = await this.syncBlockchain()
+    const walletChainId = chainId ? `carbon_${chainId.toString()}-1` : ''
+    const api = this.getApi()
+    if (walletChainId !== evmChainId) {
+        memo += "|CROSSCHAIN-SIGNING|signed-chain-id:" + walletChainId + ";" + "carbon-chain-id:" + evmChainId
+    }
+    const stdSignDoc = makeSignDoc(msgs, fee, walletChainId || evmChainId, memo, accountNumber, sequence)
```

Comment:
same discussion as above, tot we dont wanna edit touch the signdoc liao

Reply:
yeah signdoc chain id wont be touched but the memo will still be appended